### DISCUSSION TITLE
MHE parameter update to MPC

### DISCUSSION THREAD

**Discussion by SR-dTdt** [do-mpc/do-mpc#143] [Q&A] [answered]
Hi,

I am using a MHE approach to update a parameter of my model. How can I transfer the updated parameter to MPC?

Best,
SR

**Comment by Filippo-Tamagnini** [comment, marked as answer]
You can access the estimated parameters via the estimator property p_est0 that is updated at each iteration, see https://www.do-mpc.com/en/latest/api/do_mpc.estimator.MHE.p_est0.html#do_mpc.estimator.MHE.p_est0

**Comment by sunlinghang** [comment]
Many thanks for the nice package. Sorry for a follow-up question on this thread. So when using mhe together with mpc we have a piece of code over loop steps, which is pretty standard.

u0 = mpc.make_step(x0)
y_next = simulator.make_step(u0)
x0 = mhe.make_step(y_next)

In mhe.make_step(), an unknown parameter is estimated and changes between 0-1. And I need to feed this parameter in mpc.make_step() as my cost function is dependent on this parameter. However, I just found that this parameter stays the initial value in mpc.make_step(). So how can i update this parameter in mpc.make_step() as model.p['par'] is also constant?

Many thanks,
LS

**Reply by sunlinghang** [reply]
Just made a workaround, my parameter is a single variable. I've added one line in the loop

u0 = mpc.make_step(x0, **_*r0*_**)
y_next = simulator.make_step(u0)
x0 = mhe.make_step(y_next)
**_*r0 = estimator.data._p[step, 0]*_**

and then changed the source code for the function mpc.make_step() by changing **_*self.opt_p_num['_p'] = r0*_**
Hope it helps if anyone faces the same issue
